File: Assets/Script/GameMaster.js
#pragma strict
var Started : boolean = false;

var Floor_Prefab : GameObject;
var Floor_New : GameObject;
var Quantity : int = 1;
var X : int = -10; 

var Intro_Text : GameObject;
var CurrentTime : UnityEngine.UI.Text;
var HighTime : UnityEngine.UI.Text;
var Time_Elapsed : int;

var Player : GameObject;
var Cameras : GameObject; // For location
var The_Camera : Rigidbody;
var Speed : float = 3;

function Start () {
	while (Quantity < 30) {
		Floor_New = Instantiate(Floor_Prefab, new Vector3 (X, 0, 0),  Quaternion.identity);
    	PlayerPrefs.SetInt("Block Total", PlayerPrefs.GetInt("Block Total") + 1);
    	Quantity++;
    	X++;
    	Floor_New.name = "Floor #" + Quantity;
    }

    if (Application.loadedLevelName == "MainMenu2") {
    	Started = true;
		Creation();
		TimeAdder();
    } 
}

function Update () {
	if (Input.GetMouseButtonDown(0) && Started == false && Time_Elapsed <= 0) {
		Started = true;
		Creation();
		TimeAdder();
	}else if (Input.GetMouseButtonDown(0) && Started == false) {
		Application.LoadLevel("MainMenu2");
	}

	if (Player.transform.position.y < -5) {
		Started = false;
	}

	if (Started == true) {
		Intro_Text.SetActive(false);
		Player.transform.Translate(Vector3.right * Speed * Time.deltaTime);
		The_Camera.velocity.x = Speed;
		Speed = Speed * 1.0001;
	}else {
		Intro_Text.SetActive(true);
		The_Camera.velocity.x = 0;
	}

	CurrentTime.text = "Time Survived: " + Time_Elapsed + " Seconds";
	if (Time_Elapsed > PlayerPrefs.GetInt("HighTime")) {
		PlayerPrefs.SetInt("HighTime", Time_Elapsed);
	}
	HighTime.text = "Longest Time: " + PlayerPrefs.GetInt("HighTime") + " Seconds";
}

function Creation () {
	while (Started == true) {
    	var Gap_Next : int = Random.Range(1, 10);
    	while (Gap_Next > 0) {
    		Floor_New = Instantiate(Floor_Prefab, new Vector3 (X, 0, 0),  Quaternion.identity);
    		Quantity++;
    		X++;
    		Gap_Next--;
    		if (Started == true) {
    			Destroy (Floor_New, 120);
    		}
    		Floor_New.name = "Floor #" + Quantity;
    		yield WaitForSeconds (.2); 
    	}
    	X = X + Random.Range(2, 4);
    }
}

function TimeAdder () {
	while (Started == true) {
		yield WaitForSeconds (1); 
		Time_Elapsed++;
	}
}

File: Assets/Script/GameMaster.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class GameMaster : MonoBehaviour
{
    public bool Started;
    public GameObject Floor_Prefab;
    public GameObject Floor_New;
    public int Quantity;
    public int X;
    public GameObject Intro_Text;
    public UnityEngine.UI.Text CurrentTime;
    public UnityEngine.UI.Text HighTime;
    public int Time_Elapsed;
    public GameObject Player;
    public GameObject Cameras; // For location
    public Rigidbody The_Camera;
    public float Speed;
    public virtual void Start()
    {
        while (this.Quantity < 30)
        {
            this.Floor_New = UnityEngine.Object.Instantiate(this.Floor_Prefab, new Vector3(this.X, 0, 0), Quaternion.identity);
            PlayerPrefs.SetInt("Block Total", PlayerPrefs.GetInt("Block Total") + 1);
            this.Quantity++;
            this.X++;
            this.Floor_New.name = "Floor #" + this.Quantity;
        }
        if (Application.loadedLevelName == "MainMenu2")
        {
            this.Started = true;
            this.StartCoroutine(this.Creation());
            this.StartCoroutine(this.TimeAdder());
        }
    }

    public virtual void Update()
    {
        if ((Input.GetMouseButtonDown(0) && (this.Started == false)) && (this.Time_Elapsed <= 0))
        {
            this.Started = true;
            this.StartCoroutine(this.Creation());
            this.StartCoroutine(this.TimeAdder());
        }
        else
        {
            if (Input.GetMouseButtonDown(0) && (this.Started == false))
            {
                Application.LoadLevel("MainMenu2");
            }
        }
        if (this.Player.transform.position.y < -5)
        {
            this.Started = false;
        }
        if (this.Started == true)
        {
            this.Intro_Text.SetActive(false);
            this.Player.transform.Translate((Vector3.right * this.Speed) * Time.deltaTime);

            {
                float _3 = this.Speed;
                Vector3 _4 = this.The_Camera.velocity;
                _4.x = _3;
                this.The_Camera.velocity = _4;
            }
            this.Speed = this.Speed * 1.0001f;
        }
        else
        {
            this.Intro_Text.SetActive(true);

            {
                int _5 = 0;
                Vector3 _6 = this.The_Camera.velocity;
                _6.x = _5;
                this.The_Camera.velocity = _6;
            }
        }
        this.CurrentTime.text = ("Time Survived: " + this.Time_Elapsed) + " Seconds";
        if (this.Time_Elapsed > PlayerPrefs.GetInt("HighTime"))
        {
            PlayerPrefs.SetInt("HighTime", this.Time_Elapsed);
        }
        this.HighTime.text = ("Longest Time: " + PlayerPrefs.GetInt("HighTime")) + " Seconds";
    }

    public virtual IEnumerator Creation()
    {
        while (this.Started == true)
        {
            int Gap_Next = Random.Range(1, 10);
            while (Gap_Next > 0)
            {
                this.Floor_New = UnityEngine.Object.Instantiate(this.Floor_Prefab, new Vector3(this.X, 0, 0), Quaternion.identity);
                this.Quantity++;
                this.X++;
                Gap_Next--;
                if (this.Started == true)
                {
                    UnityEngine.Object.Destroy(this.Floor_New, 120);
                }
                this.Floor_New.name = "Floor #" + this.Quantity;
                yield return new WaitForSeconds(0.2f);
            }
            this.X = this.X + Random.Range(2, 4);
        }
    }

    public virtual IEnumerator TimeAdder()
    {
        while (this.Started == true)
        {
            yield return new WaitForSeconds(1);
            this.Time_Elapsed++;
        }
    }

    public GameMaster()
    {
        this.Quantity = 1;
        this.X = -10;
        this.Speed = 3;
    }

}